### PR TITLE
Updated FCS to .NET/F# 9.0 + Prepare Fable 5 alpha

### PR DESCRIPTION
**Pull request by ncave**
- Updated FCS to .NET/F# 9.0 (commit [3dff60a](https://github.com/dotnet/fsharp/tree/3dff60a1dbfca054177ab13f3e4e2353a66d25c2))
- Changed Analyzers build image to `ubuntu-latest`
- Updated Fable TFM to .NET 8.0
- Updated build dependencies
- Removed rarely used build dependencies

### CODE REVIEW

**Comment by ncave** [comment]
@MangelMaxime This is just a first pass. We may have to update the Fable CLI to .NET 8.0, now that .NET 6.0 is no longer supported. There are quite a few security warning for old dependencies that need to be updated.

Before we go there, do you mind publishing a new release with all the pending changes that have already been merged?

**Comment by ncave** [comment]
Here is why the Analyzers build fails: https://github.com/dotnet/sdk/issues/44868
We may have to disable that until it's resolved.

**Comment by MangelMaxime** [comment]
Hello @ncave,

I made a new Fable release with all the changes available on `main`.

However, to do the release I had to revert [Run husky with --allow-roll-forward](https://github.com/fable-compiler/Fable/pull/3956) because I had an error telling me that `--allow-roll-forward` is not a recognised argument.

I didn't investigate any further as I wanted to make the release and we can re-introduce `--allow-roll-forward` if needed.

**Comment by ncave** [comment]
@MangelMaxime 
> I had to revert https://github.com/fable-compiler/Fable/pull/3956

That's fine, I'll add it here.

**Comment by ncave** [comment]
@MangelMaxime 
Rebased, ready for review. Let me know if you think we need to bump the Fable major version for this change in TFM.

**Comment by MangelMaxime** [comment]
@ncave Thank you, I will have a look at it tomorrow.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -222,37 +222,39 @@ jobs:
 
   # Separate job to run F# analyzers
   analyzers:
-      runs-on: windows-latest
+    runs-on: ubuntu-latest
 
-      steps:
-      - uses: actions/checkout@v4
+    steps:
+    - uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (ncave) [reply .github/workflows/build.yml]: Not intentional, just copy-paste. Fixed.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -222,37 +222,39 @@ jobs:
...
 
-      - name: Setup .NET
-        uses: actions/setup-dotnet@v4
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v1
```
[the remark this answers is not in the document]
Comment (ncave) [reply .github/workflows/build.yml]: Not intentional, just copy-paste. Fixed.

**Comment by MangelMaxime** [comment]
I think this is ready to be merge but we need to make a decision regarding the version number indeed.

I believe we should probably bump the major version because people requesting .NET 6 using `global.json` will not be able to use Fable 5. At least, that's what I would expect dotnet/NuGet to do.

If we do the bump, here are the "side effects" we should resolve (perhaps there are some others too):

1. Can this cause issues with Fable compiler plugins?

	In theory, this should be fine because Compiler plugins should have defined something like `override _.FableMinimumVersion = "4.0"` 
	
	[Example with Feliz](https://github.com/Zaid-Ajaj/Feliz/blob/6841c9b150db7d73f1c9549ede07964949b5eaf0/Feliz.CompilerPlugins/ReactComponent.fs#L58C5-L58C43)

	There are probably only 2 plugin really impact by that Feliz which should be ok and [Oxpecker.Solid](https://github.com/Lanayx/Oxpecker/tree/main/src/Oxpecker.Solid) which should also be fine.

2. What about Fable.AST ? 

	I think I remember a discussion where we mentioned that Fable AST version was mirroring Fable version. So Fable AST 4.x was for Fable 4.

	Should we also bump Fable.AST to version 5 or says that Fable.AST 4 is compatible with Fable 4 & 5?
	
	I think the easier path is to consider Fable.AST 4 compatible with Fable 4 & 5. We can even make a matrix in Fable documentation to make an official statement.

3. If we bump to .NET 8, it means we are now in a situation where we can make `MSBuildCrackerResolver` the default instead of `BuildalyzerCrackerResolver`.

	If we do the major bump, I think this a good opportunity to do the switch and add a `--legacy-resolver` options to Fable CLI as a potential workaround if something was not catch during the experimentation phase of `MSBuildCrackerResolver`.

4. We need to archive current Fable REPL under `/repl4` and make the `/repl` target Fable 5.

5. We need to create `FABLE_COMPILER_5` compiler directive. There is a risk that some libraries will need to be updated if they used `#if FABLE_COMPILER_4` but I expect this impact to be minimal.

6. I checked SAFE Stack template and they are already using .NET 8 via `global.json`. So at least, here we should avoid too much migration path issues if we bump to a .NET 8 tool.

With all that said I see 2 solutions:

1. We keep release F# 9 support in Fable 4 targeting .NET 6, which will give us time to think about the issues mentioned above. We can 
2. We go with Fable 5 and decide now what happens in regards of the different impacted areas.

**Comment by ncave** [comment]
@MangelMaxime All good points, I fully agree (especially #3, I was going to say the same thing).
IMO, let's just go with it, the nullability change in F# 9 is reason enough to bump major version in Fable too.

**Comment by nojaf** [comment]
My two cents: go with a major and publish alpha's until you get it right.

**Comment by MangelMaxime** [comment]
You are right @nojaf better be safe than sorry.

@ncave Unless, you started working on the different items I listed, I can work on them and if all goes well make an alpha version to check it on a "real project".

**Comment by ncave** [comment]
@MangelMaxime Please go ahead.

**Comment by MangelMaxime** [comment]
@ncave Fable 5.0.0-alpha.1 has been released.

I made the `main` branch for Fable 5 release and created a `fable4` branch if we need to make a release for this version.

I did it in this order because, I don't see Fable 5 to stay in alpha for a long time and like that we can focus on a single version instead of maintaining 2 versions at the same time. If needed, we can always revisit this position.

**Pull request merged**